src/librustc_unicode/char.rs

```diff
@@ -406,9 +572,22 @@ impl char {
         }
     }
 
-    /// Indicates whether a character is whitespace.
+    /// Returns true if this `char` is whitespace, and false otherwise.
+    ///
+    /// 'Whitespace' is defined according to the terms of the Unicode Derived Core
+    /// Property `White_Space`.
+    ///
+    /// # Examples
     ///
-    /// Whitespace is defined in terms of the Unicode Property `White_Space`.
+    /// Basic usage:
+    ///
+    /// ```
+    /// let c = ' ';
+    /// assert!(c.is_whitespace());
```

Comment:
Please add an example with `'\u{A0}'` (the non-breaking space), to show that this is not ASCII whitespace, and using escaping rather than a literal `' '` which looks identical to the `' '` ASCII space.